scheduler/scheduler.go

```diff
@@ -171,6 +194,15 @@ func (w *worker) k8sify(
 	}
 
 	containerCount := len(podSpec.Containers) + systemContainers
+
+	for i, c := range k8sPlugin.Sidecars {
+		if c.Name == "" {
```

Comment:
have we tested what happens here if there are conflicts in the names? I guess that would be a failure to schedule, but can we work around that somehow?

Reply:
Created a separate issue to think more about this kind of thing: https://github.com/buildkite/agent-stack-k8s/issues/111